### PR TITLE
feat(theming): add global theming overrides

### PR DESCRIPTION
**Pull request by tw15egan**
### Added
3 new variables
- `$input-border`: this controls the border on input elements (`dropdown`, `select`, `number-input`, `text-input`, `textarea`). It is `1px solid transparent` by default.
- `$input-label-weight`: controls the font-weight of the `.bx--label` class. It is set to 700 by default. 
- `$css--use-layer`: A global flag to control whether or not layer is applied to all elements. It is set to `true` by default.

### Changed
- Layer mixin has been updated to check if the global `$css--use-layer` flag has been to set to false. If it has, it does not generate any box-shadow.
- Above mentioned input elements to use `$input-border` instead of the hard coded value.
- `bx--label` now uses `$input-label-weight` to set it's font-weight

### CODE REVIEW

File: src/globals/scss/styles.scss [hunk 1]
```diff
@@ -6,6 +6,7 @@ $css--font-face: false !default;
 $css--helpers: true !default;
 $css--body: true !default;
 $css--reset: false !default;
+$css--use-layer: true !default;
```
Comment (hellobrian) [inline src/globals/scss/styles.scss:9]: It might look less "consistent" but I don't think this variable should be using `css` in its name. 
The reason why all the other variables use `css` in their names is because they output selectors and CSS rules.

Maybe we can name this `$global--use-layer`?

File: src/globals/scss/_layer.scss [hunk 2]
```diff
@@ -35,10 +35,12 @@ $layer-shadows: (
 
 // Layer mixin
 @mixin layer($layer) {
-  @if map-has-key($layer-shadows, $layer) {
-    box-shadow: #{map-get($layer-shadows, $layer)};
-  }
-  @else {
-    @warn '#{$layer} is not a valid layer.';
+  @if ($css--use-layer) {
```
Comment (hellobrian) [inline src/globals/scss/_layer.scss]:
```sh
Error in plugin 'sass'
Message:
    src/globals/scss/_layer.scss
Error: Undefined variable: "$css--use-layer".
        on line 38 of src/globals/scss/_layer.scss
>>   @if ($css--use-layer) {
```

This mixin forces the user to always have the variable declared, otherwise they get an undefined variable error.
You can match the [conditional logic we have for other global variables](https://github.com/carbon-design-system/carbon-components/pull/76/files#diff-02222be8ee9618c8b4dbd6a9a584d102R29). (PR #76) 

```scss
@if global-variable-exists('css--use-layer') == false or $css--use-layer == true {
  @include layer($layer); // layer mixin can remove if/else statement and warning and just keep box-shadow css
}
```

__Here's how it works:__ the `global-variable-exists('css--use-layer')` SCSS function checks ["whether a variable with the given name exists in the global scope (at the top level of the file)."](http://sass-lang.com/documentation/Sass/Script/Functions.html#global_variable_exists-instance_method). 

- __If the variable does not exist__: the layer mixin will get invoked.
- __If the variable exists__: then we check the boolean value of the variable.
  - __If variable is true__: invoke the mixin
  - __If variable is false__: do nothing.

__Basically:__ The mixin always gets invoked unless the user explicitly sets the global variable to `false`

File: src/globals/scss/styles.scss (hunk 1 above)
Comment (tw15egan) [reply src/globals/scss/styles.scss:9]: Hmm, maybe I'm not following, but wouldn't this flag prevent code from being output, so it *would* belong here? 

If this flag is set to false, no layer styles are added and no CSS styles are generated. If it is set to true, all layer styles will produce CSS styles

Comment (hellobrian) [reply src/globals/scss/styles.scss:9]: IMO, It's the naming that needs to change. 

For example, $css--body outputs CSS with a selector and styles:
```css
  h1 {
    @include reset;
    @include font-size('32');
    @include line-height('heading');
    @include letter-spacing;
 }
```

But `$css--use-layer` is only going to output styles and no selector:

```css
@mixin layer($layer) {
  @if global-variable-exists('css--use-layer') == false or $css--use-layer == true  {
    @if map-has-key($layer-shadows, $layer) {
      box-shadow: #{map-get($layer-shadows, $layer)};
    }
    @else {
      @warn '#{$layer} is not a valid layer.';
    }
  }
}
```

Another way I can think about this is that maybe this is a nuance that doesn't need to be reflected in the name. If we keep the variable name as `$css--use-layer`, we can just explain in the docs that all `$css--` variables are global variables for configuration of SCSS styles.

Comment (hellobrian) [reply src/globals/scss/styles.scss:9]: Just settled this offline: `v8` we should change the names from `$css--` to `$global--`.
For now, we keep the `$css--use-layer` name for consistency. All the FEDs agreed that the selector nuance is not the focus for distinction. What's more important is the output styles showing up or not.

**Review by hellobrian — APPROVED** [review]

**Comment by marijohannessen** [comment]
👍 ✅
